src/python/pants/engine/internals/graph.py

```diff
@@ -198,34 +200,18 @@ class Owners:
 async def find_owners(owners_request: OwnersRequest) -> Owners:
     sources_set = FrozenOrderedSet(owners_request.sources)
     dirs_set = FrozenOrderedSet(os.path.dirname(source) for source in sources_set)
-
-    # Walk up the buildroot looking for targets that would conceivably claim changed sources.
     candidate_specs = tuple(AscendantAddresses(directory=d) for d in dirs_set)
-    candidate_targets = await Get[HydratedTargets](AddressSpecs(candidate_specs))
-
-    # Match the source globs against the expanded candidate targets.
-    def owns_any_source(legacy_target: HydratedTarget) -> bool:
-        """Given a `HydratedTarget` instance, check if it owns the given source file."""
-        target_kwargs = legacy_target.adaptor.kwargs()
-
-        # Handle `sources`-declaring targets.
-        # NB: Deleted files can only be matched against the 'filespec' (ie, `PathGlobs`) for a target,
```

Comment:
The NB in this comment is still useful; the TODO is stale though: it is now implemented in rust.